src/scripts/eastern_kingdoms/elwynn_forest/elwynn_forest.cpp

```diff
@@ -14,80 +14,149 @@
 
-#define SAY_HEAL    -1000187
+    constexpr uint32 SAY_HEAL{ 2283 };
+    constexpr uint32 SPELL_SYMBOL_OF_LIFE{ 8593 };
 
-struct npc_henze_faulkAI : public ScriptedAI
-{
-    uint32 lifeTimer;
-    bool spellHit;
+    enum State
+    {
+        STATE_RESET = 0,
+        STATE_JUST_REVIVED,
+        STATE_DO_EMOTE
+    };
+}
 
-    npc_henze_faulkAI(Creature* pCreature) : ScriptedAI(pCreature)
+struct tome_of_divinityAI : public ScriptedAI
+{
+    explicit tome_of_divinityAI(Creature* pCreature) : ScriptedAI(pCreature)
     {
-        Reset();
+        tome_of_divinityAI::Reset();
     }
 
+    uint8 m_uiState{};
+
+    uint32 m_uiResetTimer{};
+    uint32 m_uiTalkTimer{};
+    uint32 m_uiEmoteTimer{};
+
+    ObjectGuid m_playerGuid{};
+
+    bool m_bSpellHit{};
+
     void Reset() override
     {
-        lifeTimer = 120000;
+        m_uiState = tome_of_divinity::STATE_RESET;
+
+        m_uiResetTimer = 120000;
+        m_uiTalkTimer = 1000;
+        m_uiEmoteTimer = 1000;
+
+        m_playerGuid.Clear();
+
+        m_bSpellHit = false;
+
         m_creature->SetUInt32Value(UNIT_DYNAMIC_FLAGS, UNIT_DYNFLAG_DEAD);
-        m_creature->SetStandState(UNIT_STAND_STATE_DEAD);   // lay down
-        spellHit = false;
+        m_creature->SetStandState(UNIT_STAND_STATE_DEAD);
+        m_creature->SetDeathState(JUST_DIED);
     }
 
-    void MoveInLineOfSight(Unit *who) override { }
-
-    void UpdateAI(uint32 const diff) override
+    void UpdateAI(const uint32 uiDiff) override
     {
         if (m_creature->IsStandingUp())
         {
-            if (lifeTimer < diff)
-                m_creature->AI()->EnterEvadeMode();
-            else
-                lifeTimer -= diff;
+            switch (m_uiState)
+            {
+            case tome_of_divinity::STATE_JUST_REVIVED:
+            {
+                if (m_uiTalkTimer < uiDiff)
+                {
+                    if (Player * pPlayer{ sObjectAccessor.FindPlayer(m_playerGuid) })
+                    {
+                        m_creature->SetFacingToObject(pPlayer);
+                        DoScriptText(tome_of_divinity::SAY_HEAL, m_creature, pPlayer);
+                    }
+
+                    m_playerGuid.Clear();
+
+                    m_uiState = tome_of_divinity::STATE_DO_EMOTE;
+                }
+                else
+                {
+                    m_uiTalkTimer -= uiDiff;
+                }
+
+                break;
+            }
+            case tome_of_divinity::STATE_DO_EMOTE:
+            {
+                if (m_uiEmoteTimer < uiDiff)
+                {
+                    m_creature->HandleEmote(EMOTE_ONESHOT_BOW);
+                    m_uiState = tome_of_divinity::STATE_RESET;
+                }
+                else
+                {
+                    m_uiEmoteTimer -= uiDiff;
+                }
+
+                break;
+            }
+            case tome_of_divinity::STATE_RESET:
+                [[fallthrough]];
```

Comment:
this is not valid in c++14
https://en.cppreference.com/w/cpp/language/attributes/fallthrough